### PR TITLE
shrink send buffer after a big query

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
ref https://github.com/pingcap/tikv/issues/1203
Shrink send buffer to default size after a big query has returned.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@BusyJay @siddontang PTAL

**Comment by BusyJay** [comment]
LGTM

File: src/server/conn.rs [hunk 1]
```diff
@@ -250,6 +251,9 @@ impl Conn {
         }
 
         // no data for writing, remove writable
+        if self.send_buffer.capacity() > SEND_BUFFER_SHRINK_THRESHOLD {
```
Comment (siddontang) [inline src/server/conn.rs]: can we add a configuration for test coverage, and the store connection can use a bigger threshold.

**Comment by siddontang** [comment]
No need to handle recv buffer?

**Comment by zhangjinpeng1987** [comment]
`No need to handle recv buffer?`
`recv_buffer` will be take away when decode message, is it? @BusyJay

**Comment by BusyJay** [comment]
No. But it reads one message then decodes it immediately, so the size of the recv_buffer will be the maximum size of messages.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**